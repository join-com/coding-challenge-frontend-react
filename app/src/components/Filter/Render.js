import React from 'react';
import PropTypes from 'prop-types';

import Input from 'app/components/Input';
import Button from 'app/components/Button';

import StyledComponent from './_style';

export default function Filter({
    handleInputChange,
    handleSubmit,
    formData
}) {
    return (
        <StyledComponent
            className="c-filter"
            onSubmit={handleSubmit}
        >
            <Input
                classNameMod="search"
                name="query"
                placeholder="Search"
                onChange={handleInputChange}
                value={formData.query}
            />
            <Input
                classNameMod="date"
                placeholder="From"
                type="date"
                name="occurred_after"
                onChange={handleInputChange}
                value={formData.occurred_after}
            />
            <Input
                classNameMod="date"
                placeholder="To"
                type="date"
                name="occurred_before"
                onChange={handleInputChange}
                value={formData.occurred_before}
            />
            <Button type="submit">Find case</Button>
        </StyledComponent>
    );
}

Filter.propTypes = {
    handleInputChange: PropTypes.func,
    handleSubmit: PropTypes.func,
    formData: PropTypes.object
};
