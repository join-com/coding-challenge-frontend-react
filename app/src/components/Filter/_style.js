import styled from 'styled-components';

export default styled.form`
    display: flex;
    margin-bottom: 20px;

    .c-input {
        margin-right: 20px;
    }
`;
