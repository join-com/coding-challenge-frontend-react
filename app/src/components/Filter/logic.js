import PropTypes from 'prop-types';
import {
    compose,
    setDisplayName,
    withHandlers,
    withState,
    setPropTypes
} from 'recompose';

const defaultState = {
    query: '',
    occurred_after: '',
    occurred_before: ''
};

export default compose(
    setDisplayName('filter'),

    setPropTypes({
        onSubmit: PropTypes.func
    }),

    withState(
        'formData',
        'setFormData',
        {...defaultState}
    ),

    withHandlers({
        reset: ({ setFormData }) => () => {
            setFormData({...defaultState});
        }
    }),

    withHandlers({
        handleInputChange: props => e => {
            const {
                setFormData
            } = props;
            const {
                name,
                value
            } = e.target;

            setFormData(state => ({
                ...state,
                [name]: value
            }));
        },

        handleSubmit: props => e => {
            e && e.preventDefault();

            const {
                onSubmit,
                formData,
                reset
            } = props;

            typeof onSubmit === 'function' && onSubmit(formData);

            reset();
        }
    })
);
