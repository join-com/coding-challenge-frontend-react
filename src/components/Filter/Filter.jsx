import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Row } from "../ui/Layout/StyledLayout";
import Input from "../ui/Input";
import Text from "../ui/Text";
import Select from "../ui/Select";
import { ITEMS_PER_PAGE_OPTIONS } from "../../constants/global";

function Filter({ onChange, value, disableInput, itemsPerPage, changeShowItemsPerPage, totalIncidentsLength }) {
  return (
    <Fragment>
      <Row>
        <Input
          disabled={disableInput}
          size="large"
          id="filter"
          name="filter"
          uppercase
          onChange={onChange}
          value={value}
        />
      </Row>
      <Row>
        <Select
          label="Show item per page: "
          onChange={changeShowItemsPerPage}
          id="item_per_page"
          name="item_per_page"
          options={ITEMS_PER_PAGE_OPTIONS}
          value={itemsPerPage}
        />
        <Text>Total: {totalIncidentsLength}</Text>
      </Row>
    </Fragment>
  );
}

Filter.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  disableInput: PropTypes.bool,
  totalIncidentsLength: PropTypes.number.isRequired
};

Filter.defaultProps = {
  value: "",
  disableInput: false
};

export default Filter;
