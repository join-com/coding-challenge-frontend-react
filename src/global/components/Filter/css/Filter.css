:root {
  /** Typography **/
  --filter-input-alert-message-font-size: 14px;
  --filter-input-alert-message-font-weight: 400;
  --filter-input-alert-message-font-style: italic;
  --filter-input-alert-message-color: var(--color-alert);
  --filter-clear-color: var(--color-black);
  --filter-clear-background-color: var(--color-cancel);
  --filter-submit-color: var(--color-white);
  --filter-submit-background-color: var(--color-brand-primary);
}

.Filter {
  padding: var(--section-padding) 0;
}

.Filter__container {
  display: flex;
  align-items: baseline;
  justify-content: space-between;
}

.FilterInputs {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.FilterActions {
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  padding-bottom: var(--section-padding);
}
