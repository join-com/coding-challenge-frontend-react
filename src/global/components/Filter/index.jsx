import React, { Component } from 'react';

import PropTypes from 'prop-types';
import filterType from './types/filter';
import filtersValuesType from './types/filtersValues';

import {
  initFiltersValues,
  formatDataFiltersParamsToValues,
  formatFiltersValuesToDataParams,
} from './utils/FilterHelpers';

import Filter from './Filter';

class FilterWrapper extends Component {
  constructor(props) {
    super(props);

    const { filters, dataFilters } = props;
    const filtersValues = {
      ...initFiltersValues(filters),
      ...formatDataFiltersParamsToValues(dataFilters, filters),
    };

    this.state = {
      filtersValues,
    };

    this.handleFiltersChange = this.handleFiltersChange.bind(this);
    this.handleClearFilters = this.handleClearFilters.bind(this);
    this.handleSubmitFilters = this.handleSubmitFilters.bind(this);
  }

  handleFiltersChange(key, value) {
    const { filtersValues } = this.state;
    const newFiltersValues = {
      ...filtersValues,
      [key]: value,
    };

    this.setState({ filtersValues: newFiltersValues });
  }

  handleSubmitFilters(event) {
    event.preventDefault();

    const { filters, fetchFilteredData } = this.props;
    const { filtersValues } = this.state;

    const data = formatFiltersValuesToDataParams(filtersValues, filters);
    fetchFilteredData(data);
  }

  handleClearFilters() {
    const { fetchFilteredData } = this.props;
    fetchFilteredData({});

    this.setState({ filtersValues: {} });
  }

  render() {
    const { filtersValues } = this.state;
    return (
      <Filter
        {...this.props}
        filtersValues={filtersValues}
        handleFiltersChange={this.handleFiltersChange}
        handleClearFilters={this.handleClearFilters}
        handleSubmitFilters={this.handleSubmitFilters}
      />
    );
  }
}

export default FilterWrapper;

FilterWrapper.propTypes = {
  filters: PropTypes.arrayOf(filterType),
  dataFilters: filtersValuesType,
  fetchFilteredData: PropTypes.func,
};

FilterWrapper.defaultProps = {
  filters: [],
  dataFilters: {},
  fetchFilteredData: () => {},
};
