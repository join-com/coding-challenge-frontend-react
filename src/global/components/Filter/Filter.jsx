import React from 'react';

import PropTypes from 'prop-types';
import filterType from './types/filter';
import filtersValuesType from './types/filtersValues';

import { bemCls } from '../../utils/ClassNameHelpers';

import FilterClearButton from './FilterClearButton';
import FilterSubmitButton from './FilterSubmitButton';
import FilterText from './FilterText';
import FilterDateRange from './FilterDateRange';

import './css/Filter.css';

const Filter = ({
  filters, filtersValues, className,
  handleFiltersChange, handleClearFilters, handleSubmitFilters,
}) => (
  <form
    name="Filter"
    className={`Filter ${className}`}
    onSubmit={handleSubmitFilters}
  >
    <div className={bemCls(`Filter ${className}`, '__container')}>
      <div className="FilterInputs">
        {filters.map((filter) => {
          const { name } = filter;

          const filterProps = {
            key: name,
            filter,
            value: filtersValues[name],
            handleFiltersChange: handleFiltersChange,
            className: bemCls(className, 'Input'),
          };

          switch (filter.type) {
            case 'date':
              return (
                <FilterDateRange
                  {...filterProps}
                  value={{
                    from: filtersValues[`${name}From`],
                    to: filtersValues[`${name}To`],
                  }}
                />
              );

            default:
              return <FilterText {...filterProps} />;
          }
        })}
      </div>

      <div className="FilterActions">
        <FilterSubmitButton
          handleSubmitFilters={handleSubmitFilters}
          className={bemCls(className, '__submitButton')}
        />

        <FilterClearButton
          handleClearFilters={handleClearFilters}
          className={bemCls(className, '__clearButton')}
        />
      </div>
    </div>
  </form>
);

export default Filter;

Filter.propTypes = {
  filters: PropTypes.arrayOf(filterType),
  filtersValues: filtersValuesType,
  className: PropTypes.string,
  handleFiltersChange: PropTypes.func,
  handleClearFilters: PropTypes.func,
  handleSubmitFilters: PropTypes.func,
};

Filter.defaultProps = {
  filters: [{}],
  filtersValues: {},
  className: '',
  handleFiltersChange: () => {},
  handleClearFilters: () => {},
  handleSubmitFilters: () => {},
};
