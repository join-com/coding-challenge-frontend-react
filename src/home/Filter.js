import React from "react";
import "../App.css";
import TextField from "@material-ui/core/TextField";
import { connect } from "react-redux";
import {
  KeyboardDatePicker,
} from "@material-ui/pickers";
import SearchIcon from "@material-ui/icons/Search";
import Button from "@material-ui/core/Button";
import { fetchAllIncidents, setFromFilter, setKeywordFilter, setToFilter } from "../api/apiClient";
import { bindActionCreators } from "redux";

const spacerStyle = { width: "16pt" };
const parentStyle = { display: "flex", alignItems: "flex-end", justifyContent: "space-between" };
const buttonStyle = { marginBottom: "8pt" };

function Filter(props) {
  function handleKeywordChange(keyword) {
    props.setKeywordFilter(keyword);
  }

  function handleFromChange(from) {
    props.setFromFilter(from);
  }

  function handleToChange(to) {
    props.setToFilter(to);
  }

  function handleFilterClick() {
    props.fetchAllIncidents({
      keyword: props.keyword,
      from: props.from,
      to: props.to
    })
  }

  return (
    <div style={parentStyle}>
      <TextField id="filter-keyword"
                 label="Filter"
                 value={props.keyword}
                 onChange={event => handleKeywordChange(event.target.value)}
                 margin="normal" />

      <div style={spacerStyle} />

      <KeyboardDatePicker margin="normal"
                          label="Date picker"
                          value={props.from ? props.from : undefined}
                          onChange={handleFromChange} />

      <div style={spacerStyle} />

      <KeyboardDatePicker margin="normal"
                          label="Date picker"
                          value={props.to ? props.to : undefined}
                          onChange={handleToChange} />

      <div style={spacerStyle} />

      <Button style={buttonStyle} variant="outlined" aria-label="Search" onClick={() => handleFilterClick()}>
        <SearchIcon />
        Search
      </Button>

    </div>

  );
}

const mapStateToProps = state => state.filter;

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchAllIncidents,
  setToFilter,
  setFromFilter,
  setKeywordFilter
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Filter);