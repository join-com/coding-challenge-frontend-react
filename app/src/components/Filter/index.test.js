import React from 'react';
import toJson from 'enzyme-to-json';

import Filter from './index';

/* global shallow, jest, mount */
describe('<Filter />', () => {
    it('should render Filter', () => {
        const wrapper = mount(<Filter />);

        expect(wrapper.find('Filter')).toHaveLength(1);
    });

    it('should submit form', () => {
        let formData;
        const handleSubmit = jest.fn(d => formData = d);
        const wrapper = mount(<Filter onSubmit={handleSubmit} />);

        wrapper.find('input[name="query"]').simulate('change', {
            target: {
                name: 'query',
                value: 'query'
            }
        });
        wrapper.find('input[name="occurred_after"]').simulate('change', {
            target: {
                name: 'occurred_after',
                value: 'occurred_after'
            }
        });
        wrapper.find('input[name="occurred_before"]').simulate('change', {
            target: {
                name: 'occurred_before',
                value: 'occurred_before'
            }
        });

        wrapper.find('form').simulate('submit');

        expect(handleSubmit).toHaveBeenCalled();
        expect(formData).toEqual({
            query: 'query',
            occurred_after: 'occurred_after',
            occurred_before: 'occurred_before'
        });
    });

    it('should matches the snapshot', () => {
        const tree = mount(<Filter />);

        expect(toJson(tree)).toMatchSnapshot();
    });
});
