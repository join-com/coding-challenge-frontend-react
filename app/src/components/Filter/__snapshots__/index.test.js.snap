// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> should matches the snapshot 1`] = `
<filter>
  <withHandlers(withHandlers(Filter))
    formData={
      Object {
        "occurred_after": "",
        "occurred_before": "",
        "query": "",
      }
    }
    setFormData={[Function]}
  >
    <withHandlers(Filter)
      formData={
        Object {
          "occurred_after": "",
          "occurred_before": "",
          "query": "",
        }
      }
      reset={[Function]}
      setFormData={[Function]}
    >
      <Filter
        formData={
          Object {
            "occurred_after": "",
            "occurred_before": "",
            "query": "",
          }
        }
        handleInputChange={[Function]}
        handleSubmit={[Function]}
        reset={[Function]}
        setFormData={[Function]}
      >
        <styled.form
          className="c-filter"
          onSubmit={[Function]}
        >
          <StyledComponent
            className="c-filter"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "lastClassName": "fdgntj",
                  "rules": Array [
                    "
    display: flex;
    margin-bottom: 20px;

    .c-input {
        margin-right: 20px;
    }
",
                  ],
                },
                "displayName": "styled.form",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "form",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onSubmit={[Function]}
          >
            <form
              className="c-filter sc-bxivhb fdgntj"
              onSubmit={[Function]}
            >
              <Input
                classNameMod="search"
                name="query"
                onChange={[Function]}
                placeholder="Search"
                value=""
              >
                <styled.input
                  className="c-input c-input--search"
                  name="query"
                  onChange={[Function]}
                  placeholder="Search"
                  value=""
                >
                  <StyledComponent
                    className="c-input c-input--search"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": true,
                          "lastClassName": "eMKLvb",
                          "rules": Array [
                            "
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
",
                          ],
                        },
                        "displayName": "styled.input",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "input",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    name="query"
                    onChange={[Function]}
                    placeholder="Search"
                    value=""
                  >
                    <input
                      className="c-input c-input--search sc-bdVaJa eMKLvb"
                      name="query"
                      onChange={[Function]}
                      placeholder="Search"
                      value=""
                    />
                  </StyledComponent>
                </styled.input>
              </Input>
              <Input
                classNameMod="date"
                name="occurred_after"
                onChange={[Function]}
                placeholder="From"
                type="date"
                value=""
              >
                <styled.input
                  className="c-input c-input--date"
                  name="occurred_after"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="From"
                  type="text"
                  value=""
                >
                  <StyledComponent
                    className="c-input c-input--date"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": true,
                          "lastClassName": "eMKLvb",
                          "rules": Array [
                            "
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
",
                          ],
                        },
                        "displayName": "styled.input",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "input",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    name="occurred_after"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="From"
                    type="text"
                    value=""
                  >
                    <input
                      className="c-input c-input--date sc-bdVaJa eMKLvb"
                      name="occurred_after"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="From"
                      type="text"
                      value=""
                    />
                  </StyledComponent>
                </styled.input>
              </Input>
              <Input
                classNameMod="date"
                name="occurred_before"
                onChange={[Function]}
                placeholder="To"
                type="date"
                value=""
              >
                <styled.input
                  className="c-input c-input--date"
                  name="occurred_before"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="To"
                  type="text"
                  value=""
                >
                  <StyledComponent
                    className="c-input c-input--date"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": true,
                          "lastClassName": "eMKLvb",
                          "rules": Array [
                            "
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
",
                          ],
                        },
                        "displayName": "styled.input",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "input",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    name="occurred_before"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="To"
                    type="text"
                    value=""
                  >
                    <input
                      className="c-input c-input--date sc-bdVaJa eMKLvb"
                      name="occurred_before"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="To"
                      type="text"
                      value=""
                    />
                  </StyledComponent>
                </styled.input>
              </Input>
              <Button
                tag="button"
                type="submit"
              >
                <styled.button
                  type="submit"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bwzfXH",
                          "isStatic": true,
                          "lastClassName": "bbfpTW",
                          "rules": Array [
                            "
    font-weight: 400;
    text-transform: uppercase;
    margin: 0;
    cursor: pointer;
    border: 0;
    background: none;
    padding: 0 10px;
    color: #fff;
    background-color: #1187d2;
",
                          ],
                        },
                        "displayName": "styled.button",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bwzfXH",
                        "target": "button",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    type="submit"
                  >
                    <button
                      className="sc-bwzfXH bbfpTW"
                      type="submit"
                    >
                      Find case
                    </button>
                  </StyledComponent>
                </styled.button>
              </Button>
            </form>
          </StyledComponent>
        </styled.form>
      </Filter>
    </withHandlers(Filter)>
  </withHandlers(withHandlers(Filter))>
</filter>
`;

exports[`<Filter /> should matches the snapshot 1`] = `
<filter>
  <withHandlers(withHandlers(Filter))
    formData={
      Object {
        "occurred_after": "",
        "occurred_before": "",
        "query": "",
      }
    }
    setFormData={[Function]}
  >
    <withHandlers(Filter)
      formData={
        Object {
          "occurred_after": "",
          "occurred_before": "",
          "query": "",
        }
      }
      reset={[Function]}
      setFormData={[Function]}
    >
      <Filter
        formData={
          Object {
            "occurred_after": "",
            "occurred_before": "",
            "query": "",
          }
        }
        handleInputChange={[Function]}
        handleSubmit={[Function]}
        reset={[Function]}
        setFormData={[Function]}
      >
        <styled.form
          className="c-filter"
          onSubmit={[Function]}
        >
          <StyledComponent
            className="c-filter"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "lastClassName": "fdgntj",
                  "rules": Array [
                    "
    display: flex;
    margin-bottom: 20px;

    .c-input {
        margin-right: 20px;
    }
",
                  ],
                },
                "displayName": "styled.form",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "form",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onSubmit={[Function]}
          >
            <form
              className="c-filter sc-bxivhb fdgntj"
              onSubmit={[Function]}
            >
              <Input
                classNameMod="search"
                name="query"
                onChange={[Function]}
                placeholder="Search"
                value=""
              >
                <styled.input
                  className="c-input c-input--search"
                  name="query"
                  onChange={[Function]}
                  placeholder="Search"
                  value=""
                >
                  <StyledComponent
                    className="c-input c-input--search"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": true,
                          "lastClassName": "eMKLvb",
                          "rules": Array [
                            "
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
",
                          ],
                        },
                        "displayName": "styled.input",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "input",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    name="query"
                    onChange={[Function]}
                    placeholder="Search"
                    value=""
                  >
                    <input
                      className="c-input c-input--search sc-bdVaJa eMKLvb"
                      name="query"
                      onChange={[Function]}
                      placeholder="Search"
                      value=""
                    />
                  </StyledComponent>
                </styled.input>
              </Input>
              <Input
                classNameMod="date"
                name="occurred_after"
                onChange={[Function]}
                placeholder="From"
                type="date"
                value=""
              >
                <styled.input
                  className="c-input c-input--date"
                  name="occurred_after"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="From"
                  type="text"
                  value=""
                >
                  <StyledComponent
                    className="c-input c-input--date"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": true,
                          "lastClassName": "eMKLvb",
                          "rules": Array [
                            "
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
",
                          ],
                        },
                        "displayName": "styled.input",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "input",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    name="occurred_after"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="From"
                    type="text"
                    value=""
                  >
                    <input
                      className="c-input c-input--date sc-bdVaJa eMKLvb"
                      name="occurred_after"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="From"
                      type="text"
                      value=""
                    />
                  </StyledComponent>
                </styled.input>
              </Input>
              <Input
                classNameMod="date"
                name="occurred_before"
                onChange={[Function]}
                placeholder="To"
                type="date"
                value=""
              >
                <styled.input
                  className="c-input c-input--date"
                  name="occurred_before"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="To"
                  type="text"
                  value=""
                >
                  <StyledComponent
                    className="c-input c-input--date"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": true,
                          "lastClassName": "eMKLvb",
                          "rules": Array [
                            "
    margin: 0;
    padding: 5px 10px;
    background-color: #fff;
    border: 1px solid #bbb;
    height: 30px;
    font-size: .9em;

    &.c-input--search {
        width: 400px;
    }

    &.c-input--date {
        width: 170px;
    }
",
                          ],
                        },
                        "displayName": "styled.input",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "input",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    name="occurred_before"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="To"
                    type="text"
                    value=""
                  >
                    <input
                      className="c-input c-input--date sc-bdVaJa eMKLvb"
                      name="occurred_before"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="To"
                      type="text"
                      value=""
                    />
                  </StyledComponent>
                </styled.input>
              </Input>
              <Button
                tag="button"
                type="submit"
              >
                <styled.button
                  type="submit"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bwzfXH",
                          "isStatic": true,
                          "lastClassName": "bbfpTW",
                          "rules": Array [
                            "
    font-weight: 400;
    text-transform: uppercase;
    margin: 0;
    cursor: pointer;
    border: 0;
    background: none;
    padding: 0 10px;
    color: #fff;
    background-color: #1187d2;
",
                          ],
                        },
                        "displayName": "styled.button",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bwzfXH",
                        "target": "button",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    type="submit"
                  >
                    <button
                      className="sc-bwzfXH bbfpTW"
                      type="submit"
                    >
                      Find case
                    </button>
                  </StyledComponent>
                </styled.button>
              </Button>
            </form>
          </StyledComponent>
        </styled.form>
      </Filter>
    </withHandlers(Filter)>
  </withHandlers(withHandlers(Filter))>
</filter>
`;
